test/api/model/events_test.dart

```diff
@@ -23,6 +23,24 @@ void main() {
     check(mkEvent([MessageFlag.read])).message.flags.deepEquals([MessageFlag.read]);
   });
 
+  test('update_message_flags/remove: require messageDetails in mark-as-unread', () {
+    final json = {
+      'id': 1,
+      'type': 'update_message_flags',
+      'op': 'remove',
+      'flag': 'starred',
+      'messages': [123],
+      'all': false,
+    };
+    check(() => UpdateMessageFlagsRemoveEvent.fromJson(json)).returnsNormally();
+    json['flag'] = 'read';
+    check(() => UpdateMessageFlagsRemoveEvent.fromJson(json)).throws();
+    json['message_details'] = {
+      '123': {'type': 'private', 'mentioned': false, 'user_ids': [2]}
+    };
+    check(() => UpdateMessageFlagsRemoveEvent.fromJson(json)).returnsNormally();
```

Comment:
This test looks good except that seeing the re-use of the `json` object makes me just a bit worried that a `fromJson` implementation might mutate the data it's passed, and confuse the test. (I'm not sure we don't already have such a `fromJson` somewhere.)

Something like `{ ...baseJson, 'flag': 'read' }` would leave me reassured. (In principle I guess the `messages` array could be mutated too, but that seems less likely and I wouldn't worry.)